third_party/closure/goog/uri/uri.js

```diff
@@ -277,7 +277,7 @@ goog.Uri.prototype.resolve = function(relativeUri) {
   }
 
   if (overridden) {
-    absoluteUri.setQueryData(relativeUri.getDecodedQuery());
+    absoluteUri.setQueryData(relativeUri.getQueryData().clone());
```

Comment:
Nice catch!  This bug appears to exist upstream, as well.  Can you report and/or patch there, too?  https://github.com/google/closure-library/blob/master/closure/goog/uri/uri.js#L335